paasta_tools/cli/cmds/push_to_registry.py

```diff
@@ -166,6 +169,57 @@ def paasta_push_to_registry(args: argparse.Namespace) -> int:
     return returncode
 
 
+def paasta_push_to_registry(args: argparse.Namespace) -> int:
+    """Upload a docker images to a registry"""
+    service = args.service
+    if service and service.startswith("services-"):
+        service = service.split("services-", 1)[1]
+    validate_service_name(service, args.soa_dir)
+
+    returncode = 0
+    image_versions = [None]
+    if args.image_version:
+        # re-tag with legacy convention as well
+        returncode = retag_versioned_image(service, args.commit, args.image_version)
+        if returncode != 0:
+            _log(
+                service=service,
+                line=f"ERROR: failed re-tagging image. See output at {get_jenkins_build_output_url()}",
+                component="build",
+                level="event",
+            )
+            return returncode
+        image_versions.append(args.image_version)
+
+    for image_version in image_versions:
+        returncode = paasta_push_to_registry_impl(args, service, image_version)
+        if returncode != 0:
+            break
+
+    return returncode
+
+
+def retag_versioned_image(service: str, commit: str, image_version: str) -> int:
+    """Create new local tag without additional version information
+
+    :param str service: service name
+    :param str commit: commit sha
+    :param str image_version: additional image version metadata
+    :return: docker command return code
+    """
+    unversioned_tag = build_docker_tag(service, commit, None)
+    image_version_tag = build_docker_tag(service, commit, image_version)
+    returncode, _ = _run(
+        ["docker", "tag", image_version_tag, unversioned_tag],
+        timeout=30,
+        log=True,
+        component="build",
+        service=service,
+        loglevel="debug",
+    )
+    return returncode
```

Comment:
I think this could lead to a little confusion down the line since the service's `make itest` or `cook-image` is really supposed to be handling all the tagging... but given we're talking about the exact same image, this seems like a clever way to minimize any jenkins-side logic!